src/System.Management.Automation/engine/Utils.cs

```diff
@@ -895,113 +895,61 @@ internal static bool IsAdministrator()
 #endif
         }
 
-        internal static bool NativeItemExists(string path)
+        internal static bool ItemExists(string path)
         {
-            bool unusedIsDirectory;
-            Exception unusedException;
+            bool result = false;
+            try
+            {
+                result = ItemExists(path, out bool unused);
+            }
+            catch
+            {
+            }
 
-            return NativeItemExists(path, out unusedIsDirectory, out unusedException);
+            return result;
         }
 
-        // This is done through P/Invoke since File.Exists and Directory.Exists pay 13% performance degradation
-        // through the CAS checks, and are terribly slow for network paths.
-        internal static bool NativeItemExists(string path, out bool isDirectory, out Exception exception)
+        internal static bool ItemExists(string path, out bool isDirectory)
```

Comment:
Similarly, maybe changing this to `CheckItemExistsAndThrowOnError`.

Reply:
I'll rename if we will not migrate to File.Exists/Directory.Exists in the PR.